### PR TITLE
Add an option to list only packages updated after a given date 

### PR DESCRIPTION
**Pull request by progval**
This allows clients to list packages updated since they last listed packages.

Additionally, as pagination is currently offset-based, it was not possible to paginate when `sort=updated_at`, as packages may be updated between two page fetches by the same client.

We would like to use this at https://www.softwareheritage.org/ to archive packages hosted on hex.pm, while skipping those with no update

### CODE REVIEW

**Comment by ericmj** [comment]
Thanks for the PR! Can you look at adding this to the `search` parameter instead? Otherwise we will have some filtering in the `search` parameter and some as individual fields.

**Comment by progval** [comment]
Like this? `/api/packages?page=1&search=updated_after:2030-01-01T00:00:00Z`

**Comment by ericmj** [comment]
Yes, correct!

File: lib/hexpm/repository/package.ex [hunk 1]
```diff
@@ -245,6 +245,14 @@ defmodule Hexpm.Repository.Package do
     from(p in query, where: description_query(p, search))
   end
 
+  defp search_param("updated_after", search, query) do
+    case DateTime.from_iso8601(search) do
+      {:ok, updated_after, 0} -> from(p in query, where: p.updated_at >= ^updated_after)
+      # invalid date, return nothing
+      _ -> from(p in query, where: false)
```
Comment (ericmj) [inline lib/hexpm/repository/package.ex:252]: In this case I think we should return 400 or ignore the filter.

**Review by ericmj — APPROVED** [review]
Looks great, just one small comment.

**Comment by wojtekmach** [comment]
Thank you!

**Pull request merged**